src/services/course-cooperation-service.ts

```diff
@@ -1,14 +1,20 @@
-import { AxiosResponse } from 'axios'
-import { URLs } from '~/constants/request'
-import { axiosClient } from '~/plugins/axiosClient'
 import { CourseCooperationResponse } from '~/types'
-import { createUrlPath } from '~/utils/helper-functions'
+import { URLs } from '~/constants/request'
+import { baseService } from '~/services/base-service'
+import { getFullUrl } from '~/utils/get-full-url'
 
 export const CoursesAndCooperationsService = {
   getByResourceId: async (
     resourceId: string
-  ): Promise<AxiosResponse<CourseCooperationResponse>> =>
-    await axiosClient.get(
-      createUrlPath(URLs.coursesAndCooperations.getByResourceId, resourceId)
-    )
+  ): Promise<CourseCooperationResponse> => {
+    const url = getFullUrl({
+      pathname: URLs.coursesAndCooperations.getByResourceId,
+      searchParameters: { resourceId }
+    })
+
+    return await baseService.request<CourseCooperationResponse>({
```

Comment:
Probably, you have no reason to await this promise here, also you can remove `async` from function declaration

```suggestion
    return baseService.request<CourseCooperationResponse>({
```

Reply:
You are absolutely right. await is not needed here because the request function already returns a promise. Thanks! Done.

Reply:
It's not completely resolved, remove `async` from functioin declaration, please